### DISCUSSION TITLE
Help with getting setup working

### DISCUSSION THREAD

**Discussion by mdeakin99** [kellerza/sunsynk#356] [Q&A]
Any ideas where I might start based on the error

Configuration

Sunsynk Inverter Dongle (192.168.1.149).  Splitter connected to CAN port.  Confirmed working as inverter is communicating with batteries.
Other port on splitter connected to https://www.amazon.co.uk/dp/B081NBCJRS?ref=ppx_yo2ov_dt_b_fed_asin_title via network cable (believe I rewired it correctly) which then connects to Home Assistant running on an odroid.
Sunsynk/Deye Inverter Add-on (multi) 
- set to pymodbus
- Tried both 192.168.1.149 and homeassistant.local in the Sunsynk/Deye Inverter Add-on (multi) configuration
- SERIAL_NR: "SerialNumberOfInverterHere"
  HA_PREFIX: SS
  MODBUS_ID: 1
  DONGLE_SERIAL_NUMBER: "0"
  PORT: tcp://homeassistant.local:502
- MQTT host/username/password match the mqtt addon/integration in HA.

Log

[20:12:12] INFO: Service restart after closing
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun sunsynk (no readiness notification)
s6-rc: info: service legacy-services successfully started
20:12:18 INFO    Loading configuration: /data/options.json
20:12:18 INFO    sunsynk library version: 0.6.5
20:12:18 INFO    Schedules:
+-----------+-----+------+--------+-----------+----------+------------+
|    Key    | src | Read | Report | Change by | Change % | Change any |
+-----------+-----+------+--------+-----------+----------+------------+
| date_time |     |  60  |   60   |           |          |    True    |
|     rw    |     |  5   |  300   |           |          |    True    |
|    enum   |     |  5   |  300   |           |          |    True    |
|     w     |  *  |  5   |   60   |     80    |          |            |
|    kwh    |     | 300  |  300   |           |          |            |
|  any_unit |     |  15  |  300   |           |          |            |
|  no_unit  |     |  15  |  300   |           |          |    True    |
+-----------+-----+------+--------+-----------+----------+------------+
20:12:18 INFO    Using Single phase sensor definitions.
20:12:18 INFO    Added hidden sensors as other sensors depend on it: Rated power, Serial
20:12:18 INFO    Connecting to tcp://homeassistant.local:502
20:12:18 INFO    PyModbus 3.7.2 tcp: homeassistant.local:502
20:12:19 WARNING Failed to connect Multiple exceptions: [Errno 111] Connect call failed ('172.30.32.1', 502), [Errno 111] Connect call failed ('172.30.232.1', 502), [Errno 111] Connect call failed ('192.168.1.102', 502), [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument, [Errno 22] Invalid argument
20:12:19 INFO    ############################################################
20:12:19 INFO        Could not connect to tcp://homeassistant.local:502:
20:12:19 INFO    ############################################################
20:12:19 CRITICAL This Add-On will terminate in 30 seconds, use the Supervisor Watchdog to restart automatically.

**Comment by mdeakin99** [comment]
I changed the port to tcp://homeassistant.local:1883 (the mqtt port) and now it seems to have go further.  I reduced Read_sensors_batch_size to 8 but still get the message below.

21:10:33 INFO    Using Single phase sensor definitions.
21:10:33 INFO    Added hidden sensors as other sensors depend on it: Rated power, Serial
21:10:33 INFO    Connecting to tcp://homeassistant.local:1883
21:10:33 INFO    PyModbus 3.7.2 tcp: homeassistant.local:1883
21:10:33 INFO    Reading startup sensors Prog2 Time, Prog5 Time, Prog6 Time, Serial, Prog3 Time, Battery Low Capacity, Rated power, Prog4 Time, Prog1 Time
21:11:17 ERROR   timeout reading 5 registers from 3; timeout reading 2 registers from 16; timeout reading 1 registers from 219; timeout reading 6 registers from 250

**Comment by mdeakin99** [comment]
Any thoughts @kellerza when you have a moment (let me see if there is a 'buy me a coffee' link for your help)

**Reply by Waz-Cpt** [reply]
I think you are quite confused as a whole on this integration.

Lets get an idea on how you have connected things up.

Where have you connected the Waveshare USB to RS485 module to?

If you are using the E-Linter type dongle, it is not used anywhere within this integration. If you are using a Solarman dongle then that is a different story.

Typically the easiest connection is when the USB to RS485 dongle is directly connected to the Home Assistant computer. You would then check which USB device it is showing up as in the all hardware section. You then use that as the connection in the configuration.

The only time you would be using an IP address is when using the Solarman dongle driver or using a support Modbus TCP gateway device or serial stream server from a customised E-Linter dongle.

**Reply by mdeakin99** [reply]
I agree… I think I am getting quite confused :)

So I have home assistant physically near the inverter.  Out of the home assistant hardware is a usb to rs485 dongle.  Out of the rs485 dongle is a wire (I took an existing Ethernet and split out the right cables) going into a splitter I purchased from solarman site and into the can bus port on the inverter (there is a separate rs485 labelled port but looking online it appears that’s not functioning).

What do you think the settings for the Sunsynk addon should look like,  in particular what should I be setting the address to tcp://localhost:8123 ?

**Reply by mdeakin99** [reply]
As an update.  I set the port to “”” and specified the usb device in debug mode to make sure it’s using the right device.

A question on the wiring for the Ethernet cable.  When it says ‘orange/white’ in the instructions,  that means the wire that’s both orange and white…. Not two wires into one connection point 

Error log below

10:22:18 INFO    Using Single phase sensor definitions.
10:22:18 INFO    Added hidden sensors as other sensors depend on it: Rated power, Serial
10:22:18 INFO    Connecting to /dev/ttyUSB0
10:22:18 INFO    PyModbus 3.7.2 Serial: /dev/ttyUSB0
10:22:18 INFO    Reading startup sensors Rated power, Prog1 Time, Prog4 Time, Prog3 Time, Serial, Prog2 Time, Prog6 Time, Battery Low Capacity, Prog5 Time
10:23:02 ERROR   timeout reading 5 registers from 3; timeout reading 2 registers from 16; timeout reading 1 registers from 219; timeout reading 6 registers from 250
10:23:46 ERROR   timeout reading 5 registers from 3; timeout reading 2 registers from 16; timeout reading 1 registers from 219; timeout reading 6 registers from 250
10:24:31 ERROR   timeout reading 5 registers from 3; timeout reading 2 registers from 16; timeout reading 1 registers from 219; timeout reading 6 registers from 250
10:24:31 WARNING Retrying individual sensors: ['rated_power', 'prog1_time', 'prog4_time', 'prog3_time', 'serial', 'prog2_time', 'prog6_time', 'battery_low_capacity', 'prog5_time']
10:24:42 ERROR   timeout reading 2 registers from 16 Rated power
10:24:53 ERROR   timeout reading 1 registers from 250 Prog1 Time
10:25:04 ERROR   timeout reading 1 registers from 253 Prog4 Time
10:25:15 ERROR   timeout reading 1 registers from 252 Prog3 Time
10:25:26 ERROR   timeout reading 5 registers from 3 Serial
10:25:37 ERROR   timeout reading 1 registers from 251 Prog2 Time
10:25:48 ERROR   timeout reading 1 registers from 255 Prog6 Time
10:26:00 ERROR   timeout reading 1 registers from 219 Battery Low Capacity
10:26:11 ERROR   timeout reading 1 registers from 254 Prog5 Time
10:26:11 CRITICAL Could not read sensors: ['Rated power', 'Prog1 Time', 'Prog4 Time', 'Prog3 Time', 'Serial', 'Prog2 Time', 'Prog6 Time', 'Battery Low Capacity', 'Prog5 Time']
10:26:11 INFO    ############################################################
10:26:11 INFO    No response on the Modbus interface /dev/ttyUSB0, see https://kellerza.github.io/sunsynk/guide/fault-finding
10:26:11 INFO    ############################################################
10:26:11 CRITICAL This Add-On will terminate in 30 seconds, use the Supervisor Watchdog to restart automatically

**Reply by Waz-Cpt** [reply]
Have you actually read through the instructions of [https://kellerza.github.io/sunsynk/](url)

You will be using the standard config.
![image](https://github.com/user-attachments/assets/b290ad2d-1e46-4b70-93b6-8c1af1118388)

It would be a good idea to check the wiring and port information provided on the SolarAssistant website.

You should have the following information set in your config
![image](https://github.com/user-attachments/assets/73425b20-9e3f-4bbb-b51b-8d911f3406c9)

Yes, have the USB RS485 device in the debug port settings as well.

**Reply by mdeakin99** [reply]
Have you actually read through the instructions > Yes but thanks for checking.

Standard config > Agreed.  This is the one I have followed.   This diagram logical rather than physical so I have have tried to interpret the bits that are not included for example ports/wiring/transport media.

Check wiring/port information > I am not using solar assistant but I will check the wiring once again, hence the question around orange/white which I asked above.  Any recommendations/tips would be greatly appreciated above and beyond 'check the wiring' as I have already taken that initial step.

Information in config > Will take a look at this.  I suspect I effectively have this given I am using the debug mode but will update and change

**Reply by mdeakin99** [reply]
Confirmed the wiring according to the instructions is correct (picture below).  Happy to be corrected.

![rg45end](https://github.com/user-attachments/assets/f1d38b5f-2a28-4d96-986b-024a7423e2e9)
![pinend](https://github.com/user-attachments/assets/5ff75c2c-fdc3-4d7b-a550-e0fafc56880f)


Update the config as you suggested but made a few additions as you actually need to put more information in for it to work it would appear.

- SERIAL_NR: "XXXXXXX"
  HA_PREFIX: SS
  MODBUS_ID: 1
  DONGLE_SERIAL_NUMBER: "0"
  PORT: /dev/ttyUSB0

Same error message around timeouts btw..

**Reply by mdeakin99** [reply]
Replacement USB device and still having the same error messages in the log eg.  

ERROR timeout reading 5 registers from 3;

**Reply by Waz-Cpt** [reply]
Strange question, are you using solid core cable or stranded?

**Reply by Waz-Cpt** [reply]
Also, which sensors have you chosen to include in the config?

Perhaps post you complete yaml config for the addon. Just blockout your MQTT details.

**Reply by mdeakin99** [reply]
> Strange question, are you using solid core cable or stranded?

Had to look that one up but it's core (the previous cable I used was stranded but changed that just in case)

**Reply by mdeakin99** [reply]
> Also, which sensors have you chosen to include in the config?
> 
> Perhaps post you complete yaml config for the addon. Just blockout your MQTT details.

I didn't pick any specifically, just went with whatever is default.  Yaml below.

DRIVER: pymodbus
INVERTERS:
  - SERIAL_NR: "1234"
    HA_PREFIX: SS
    MODBUS_ID: 1
    PORT: /dev/ttyACM0
SENSOR_DEFINITIONS: single-phase
SENSORS:
  - energy_management
  - power_flow_card
  - pv2_power
SENSORS_FIRST_INVERTER:
  - settings
MANUFACTURER: sunsynkdirect
READ_ALLOW_GAP: 2
READ_SENSORS_BATCH_SIZE: 2
SCHEDULES:
  - KEY: W
    READ_EVERY: 5
    REPORT_EVERY: 60
    CHANGE_ANY: false
    CHANGE_BY: 80
    CHANGE_PERCENT: 0
NUMBER_ENTITY_MODE: auto
PROG_TIME_INTERVAL: "60"
MQTT_HOST: core-mosquitto
MQTT_PORT: 1883
MQTT_USERNAME: mqtt_user
MQTT_PASSWORD: xxxxx

**Reply by Waz-Cpt** [reply]
So, the /dev/ttyACM0 is the same as the USB device that shows up in the debug section?

**Reply by mdeakin99** [reply]
This is as it appears in debug.

![IMG_9621](https://github.com/user-attachments/assets/02972305-3320-42c6-b9bd-143476ed90e9)

When I change the port to "" to use debug then it uses ttyACM0 but still has the same timeouts 


![IMG_9622](https://github.com/user-attachments/assets/dba9ad73-8b3a-471a-a21c-434796fba5a4)

**Reply by kellerza** [reply]
Try to install the mbusd addon, then pick a USB device there.

Then the Sunsynk addon connects to `PORT: tcp://homeassistant.local:502` or use your HASS IP, i.e. `PORT: tcp://192.168.1.231:502`

If mbusd does not connect, or you still have timeouts in the addon, it is either your wiring (which seems ok) or the port/splitter. See the docs about the splitter & the fault-finding section. 

To test the port, you can always use a laptop with a modbus app (and a short wire). 

There is really not much we can help with if the wiring does not work.

**Reply by Waz-Cpt** [reply]
There is a project called open modscan which you can use on the PC to monitor the modbus traffic from your PC with the USB-RS485 adaptor.

Is the cable using stranded core or solid core. When using stranded please use ferrules. When using solid core you should remove the resin coating to ensure proper electrical contact.

**Reply by mdeakin99** [reply]
Tried modbus and I get the following.  Slightly different start to the message but looks like the same issue.  I will try modbus app to see how I get on.

The splitter I purchased from solarassistant site.  The part of the splitcable that goes to the batteries is still working ok so guessing it isnt the splitter. Will take a look at the fault finding.
 

14:12:24 INFO    Using Single phase sensor definitions.
14:12:24 INFO    Added hidden sensors as other sensors depend on it: Rated power, Serial
14:12:24 INFO    Connecting to tcp://homeassistant.local:502
14:12:24 INFO    PyModbus 3.7.2 tcp: homeassistant.local:502
14:12:24 INFO    Reading startup sensors Prog6 Time, Prog4 Time, Battery Low Capacity, Rated power, Serial, Prog5 Time, Prog2 Time, Prog3 Time, Prog1 Time
14:12:41 ERROR   OSError reading 2 registers from 3: failed to read register 3 - function code: 131; OSError reading 2 registers from 5: failed to read register 5 - function code: 131; OSError reading 1 registers from 7: failed to read register 7 - function code: 131; OSError reading 2 registers from 16: failed to read register 16 - function code: 131; OSError reading 1 registers from 219: failed to read register 219 - function code: 131; OSError reading 2 registers from 250: failed to read register 250 - function code: 131; OSError reading 2 registers from 252: failed to read register 252 - function code: 131; OSError reading 2 registers from 254: failed to read register 254 - function code: 131
14:12:59 ERROR   OSError reading 2 registers from 3: failed to read register 3 - function code: 131; OSError reading 2 registers from 5: failed to read register 5 - function code: 131; OSError reading 1 registers from 7: failed to read register 7 - function code: 131; OSError reading 2 registers from 16: failed to read register 16 - function code: 131; OSError reading 1 registers from 219: failed to read register 219 - function code: 131; OSError reading 2 registers from 250: failed to read register 250 - function code: 131; OSError reading 2 registers from 252: failed to read register 252 - function code: 131; OSError reading 2 registers from 254: failed to read register 254 - function code: 131
14:13:16 ERROR   OSError reading 2 registers from 3: failed to read register 3 - function code: 131; OSError reading 2 registers from 5: failed to read register 5 - function code: 131; OSError reading 1 registers from 7: failed to read register 7 - function code: 131; OSError reading 2 registers from 16: failed to read register 16 - function code: 131; OSError reading 1 registers from 219: failed to read register 219 - function code: 131; OSError reading 2 registers from 250: failed to read register 250 - function code: 131; OSError reading 2 registers from 252: failed to read register 252 - function code: 131; OSError reading 2 registers from 254: failed to read register 254 - function code: 131
14:13:16 WARNING Retrying individual sensors: ['prog6_time', 'prog4_time', 'battery_low_capacity', 'rated_power', 'serial', 'prog5_time', 'prog2_time', 'prog3_time', 'prog1_time']
14:13:18 ERROR   OSError reading 1 registers from 255: failed to read register 255 - function code: 131 Prog6 Time
14:13:21 ERROR   OSError reading 1 registers from 253: failed to read register 253 - function code: 131 Prog4 Time
14:13:23 ERROR   OSError reading 1 registers from 219: failed to read register 219 - function code: 131 Battery Low Capacity
14:13:25 ERROR   OSError reading 2 registers from 16: failed to read register 16 - function code: 131 Rated power
14:13:32 ERROR   OSError reading 2 registers from 3: failed to read register 3 - function code: 131; OSError reading 2 registers from 5: failed to read register 5 - function code: 131; OSError reading 1 registers from 7: failed to read register 7 - function code: 131 Serial
14:13:34 ERROR   OSError reading 1 registers from 254: failed to read register 254 - function code: 131 Prog5 Time
14:13:36 ERROR   OSError reading 1 registers from 251: failed to read register 251 - function code: 131 Prog2 Time
14:13:38 ERROR   OSError reading 1 registers from 252: failed to read register 252 - function code: 131 Prog3 Time
14:13:40 ERROR   OSError reading 1 registers from 250: failed to read register 250 - function code: 131 Prog1 Time
14:13:40 CRITICAL Could not read sensors: ['Prog6 Time', 'Prog4 Time', 'Battery Low Capacity', 'Rated power', 'Serial', 'Prog5 Time', 'Prog2 Time', 'Prog3 Time', 'Prog1 Time']
14:13:40 INFO    ############################################################
14:13:40 INFO    No response on the Modbus interface tcp://homeassistant.local:502, see https://kellerza.github.io/sunsynk/guide/fault-finding
14:13:40 INFO    ############################################################
14:13:40 CRITICAL This Add-On will terminate in 30 seconds, use the Supervisor Watchdog to restart automatically.

**Reply by mdeakin99** [reply]
> There is a project called open modscan which you can use on the PC to monitor the modbus traffic from your PC with the USB-RS485 adaptor.
> 
> Is the cable using stranded core or solid core. When using stranded please use ferrules. When using solid core you should remove the resin coating to ensure proper electrical contact.

Removed the coating on the cable to make sure it's only the wire that connecting to the points on the waveshare.


Downloaded MODSCAN64 on my laptop.  I connected an ethernet cable to the laptop and the splitter at the other end of the cable.  I tried a few settings but couldnt get it to connect.  To be honest though,  not sure what to put in 'Connect Using' in the modbus app.  I tried both COM1 and Remote modBus TCP Server (I put in the IP address of the sunsynk dongle but pretty sure this is the wrong thing.   Any pointers here greatly appreciated.

**Reply by mdeakin99** [reply]
Would a solarman dongle perhaps fix the issue if the splitter/cable/waveshare is currently the problem? I appreciate it means I lose remote access via Sunsynk app but I can work with that

**Reply by kellerza** [reply]
Just use a laptop with a modbus app & short wire. First simplify things

**Reply by mdeakin99** [reply]
Network cable directly from laptop to inverter correct? or have I understood incorrectly

**Reply by Tabraize** [reply]
Did you end up resolving this as i am having the same issue.
